tests/lib/config.js

```diff
@@ -1055,6 +1055,205 @@ describe("Config", () => {
+                    rules: {
+                        quotes: [2, "single"]
+                    }
+                };
+
+                const actual = config.getConfig(targetPath);
+
+                assertConfigsEqual(actual, expected);
+            });
+
+            it("should merge override config when the pattern matches the file path relative to the config file", () => {
+                const config = new Config({ cwd: process.cwd() });
+                const targetPath = getFakeFixturePath("overrides", "child", "child-one.js");
+                const expected = {
+                    rules: {
+                        quotes: [2, "single"]
+                    }
+                };
+                const actual = config.getConfig(targetPath);
+
+                assertConfigsEqual(actual, expected);
+            });
+
+            it("should not merge override config when the pattern matches the absolute file path", () => {
+                const targetPath = getFakeFixturePath("overrides", "bar.js");
+                const resolvedPath = path.resolve(__dirname, "..", "fixtures", "config-hierarchy", "overrides", "bar.js");
+                const config = new Config({
+                    cwd: process.cwd(),
+                    baseConfig: {
+                        overrides: [{
+                            files: resolvedPath,
+                            rules: {
+                                quotes: [2, "single"]
+                            }
+                        }],
+                        useEslintrc: false
+                    }
+                });
+                const expected = {
+                    rules: {
+                        quotes: [2, "double"]
+                    }
+                };
+                const actual = config.getConfig(targetPath);
+
+                assertConfigsEqual(actual, expected);
+            });
+
+            it("should not merge override config when the pattern traverses up the directory tree", () => {
+                const targetPath = getFakeFixturePath("overrides", "bar.js");
+                const parentPath = "overrides/../**/*.js";
+
+                const config = new Config({
+                    cwd: process.cwd(),
+                    baseConfig: {
+                        overrides: [{
+                            files: parentPath,
+                            rules: {
+                                quotes: [2, "single"]
+                            }
+                        }],
+                        useEslintrc: false
+                    }
+                });
+                const expected = {
+                    rules: {
+                        quotes: [2, "double"]
+                    }
+                };
+                const actual = config.getConfig(targetPath);
+
+                assertConfigsEqual(actual, expected);
+            });
+
+            it("should merge all local configs (override and non-override) before non-local configs", () => {
```

Comment:
Can you add a test where a config is in `overrides` in a parent directory, but not in `overrides` in a child directory?

For example:

```yaml
# root .eslintrc.yml

overrides:
  files: 'child-dir/**/*.js'
  config:
    rules:
      semi: [error, always]
```

```yml
# child-dir/.eslintrc.yml

rules:
  semi: [error, never]
```